mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -105,7 +121,10 @@ def search(
             # Get top-k values
             cos_scores_top_k_values, cos_scores_top_k_idx = torch.topk(
                 cos_scores,
-                min(top_k + 1, len(cos_scores[1])),
+                min(
+                    top_k + 1,
+                    len(cos_scores[1]) if len(cos_scores) > 1 else len(cos_scores[-1]),
```

Comment:
If there is only one query then this line fails -- my code evaluates the instruction task query by query, so this allows for the one query case as well.